examples/custom-server/server.js

```diff
@@ -0,0 +1,33 @@
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+const accepts = require('accepts')
+
+const app = next('.', { dev: true })
+const handle = app.getRequestHandler()
+
+app.prepare()
+.then(() => {
+  createServer((req, res) => {
+    const accept = accepts(req)
+    const type = accept.type(['json', 'html'])
+    if (type === 'json') {
+      handle(req, res)
+      return
+    }
+
+    const { pathname, query } = parse(req.url, true)
+
+    if (pathname === '/a') {
+      app.render(req, res, '/b', query)
+    } else if (pathname === '/b') {
+      app.render(req, res, '/a', query)
+    } else {
```

Comment:
In addition to this basic switching, I think the example should also explicitly show how to make a parameterized route (e.g. `/blog/:id`). IMHO, this will mitigate tons of questions/confusion.

This is awesome btw!